2006-01-21  Aurelien Jarno  <aurel32@debian.org> 

	* Makeconfig: new configuration variable rtlddir to change the
	  dynamic linker in the ELF binaries, which defaults to slibdir
	  
2005-08-30  Andreas Jochen  <Andreas Jochens>
	* Makeconfig: use rtlddir instead of slibdir
	* elf/Makefile: likewise

Index: Makeconfig
===================================================================
--- Makeconfig.orig	2007-10-23 08:50:33.000000000 -0400
+++ Makeconfig	2007-10-23 08:50:37.000000000 -0400
@@ -201,6 +201,11 @@
 endif
 inst_slibdir = $(install_root)$(slibdir)
 
+# Directory of the dynamic linker in the ELF binaries
+ifndef rtlddir
+rtlddir := $(slibdir)
+endif
+
 # Prefix to put on files installed in $(libdir).  For libraries `libNAME.a',
 # the prefix is spliced between `lib' and the name, so the linker switch
 # `-l$(libprefix)NAME' finds the library; for other files the prefix is
@@ -464,7 +469,7 @@
 endif
 ifndef config-LDFLAGS
 ifeq (yesyes,$(build-shared)$(elf))
-config-LDFLAGS = -Wl,-dynamic-linker=$(slibdir)/$(rtld-installed-name)
+config-LDFLAGS = -Wl,-dynamic-linker=$(rtlddir)/$(rtld-installed-name)
 endif
 endif
 ifndef link-libc
Index: elf/Makefile
===================================================================
--- elf/Makefile.orig	2007-10-23 08:49:46.000000000 -0400
+++ elf/Makefile	2007-10-23 08:50:37.000000000 -0400
@@ -315,7 +315,7 @@
 	  | awk '($$7 ~ /^UND(|EF)$$/ && $$1 != "0:" && $$4 != "REGISTER") { print; p=1 } END { exit p != 0 }'
 
 # interp.c exists just to get this string into the libraries.
-CFLAGS-interp.c = -D'RUNTIME_LINKER="$(slibdir)/$(rtld-installed-name)"' \
+CFLAGS-interp.c = -D'RUNTIME_LINKER="$(rtlddir)/$(rtld-installed-name)"' \
 		  -DNOT_IN_libc=1
 $(objpfx)interp.os: $(common-objpfx)config.make
 
